### ISSUE TITLE
[CT-3472] [Regression] There is no way to change the target path for `dbt ls` command after the parameter was deprecated.

### ISSUE THREAD

**Issue by arbatskiy** [dbt-labs/dbt-core#9245]
### Is this a regression in a recent version of dbt-core?

- [X] I believe this is a regression in dbt-core functionality
- [X] I have searched the existing issues, and I could not find an existing issue for this regression

### Current Behavior

The `target-path` parameter was deprecated, but unfortunately, other alternatives do not function with the `dbt ls `command:

- The `DBT_TARGET_PATH` is disregarded when using this command.
- The command-line parameter `--target-path` is not valid for this command.

### Expected/Previous Behavior

**Previously**, the command used to derive the target path from the dbt_project.yaml file, allowing for path configuration.

**Expected behavior** should align with other commands, which retrieve the target path from either the environment variable `DBT_TARGET_PATH` and/or the command-line parameter `--target-path`.

### Steps To Reproduce

1. Ensure that the `dbt_profile.yaml` does not contain the target-path parameter.
1. Execute `dbt ls` and observe that the `target` is created in the project directory as expected.
1. Set `DBT_TARGET_PATH` to a directory different from the project directory; run `dbt ls` and observe that the `target` directory is still created in the project directory.
1. When you run `dbt ls --target-path <path>`, you will encounter an error indicating that the parameter is invalid.

### Environment

```markdown
- OS: MacOs
- Python: 3.9
- dbt (working version): < 1.5
- dbt (regression version): 1.5, 1.6
```

### Which database adapter are you using with dbt?

bigquery

**Comment by jtcohen6** [comment]
@arbatskiy It looks like this was solved in v1.6:
- https://github.com/dbt-labs/dbt-core/pull/7647

Have you tried running with v1.6 or v1.7?

Given that v1.5 was released back in April, and we've released two minor versions since then which the fix since then, we're unlikely to backport a patch for this to v1.5

**Comment by arbatskiy** [comment]
Thank you, can confirm it is working in `dbt=1.6.5`
